### PR TITLE
[ASCII-1330] Fix auth token global variable race

### PR DESCRIPTION
**Pull request by pgimalac**
### What does this PR do?

Wrap the auth token global variable in a `RWLock` to avoid races.

### Motivation

The auth token global variable was not protected for concurrent access, it was only initialized in the main of each binaries until now, but it started racing in tests, and it could change in the future anyway, so having a lock is safer.

### Describe how to test/QA your changes

Change should be covered by the CI and usual deployments.
If no race is reported anymore then this is fixed

### CODE REVIEW

File: pkg/api/util/util.go [hunk 1]
```diff
@@ -11,19 +11,24 @@ import (
...
+	tokenLock sync.RWMutex
+	token     string
+	dcaToken  string
 )
 
 // SetAuthToken sets the session token
 // Requires that the config has been set up before calling
 func SetAuthToken(config model.Reader) error {
+	tokenLock.Lock()
+	defer tokenLock.Unlock()
```
Comment (dustmop) [inline pkg/api/util/util.go:30]: Could you remove the comment a few lines down that says "token is only set once, no need to mutex protect".

File: pkg/api/util/util.go [hunk 2]
```diff
@@ -38,6 +43,9 @@ func SetAuthToken(config model.Reader) error {
 // CreateAndSetAuthToken creates and sets the authorization token
 // Requires that the config has been set up before calling
 func CreateAndSetAuthToken(config model.Reader) error {
+	tokenLock.Lock()
+	defer tokenLock.Unlock()
```
Comment (dustmop) [inline pkg/api/util/util.go:47]: Same here, lower down in this function.

File: pkg/api/util/util.go [hunk 3]
```diff
@@ -51,12 +59,17 @@ func CreateAndSetAuthToken(config model.Reader) error {
...
+	tokenLock.RLock()
+	defer tokenLock.RUnlock()
 	return token
 }
 
 // InitDCAAuthToken initialize the session token for the Cluster Agent based on config options
 // Requires that the config has been set up before calling
 func InitDCAAuthToken(config model.Reader) error {
+	tokenLock.Lock()
+	defer tokenLock.Unlock()
```
Comment (dustmop) [inline pkg/api/util/util.go:71]: Please also remove the comment from this function about "dcaToken".

**Review by dustmop — APPROVED** [review]
Requesting some comment updates, but LGTM after that

**Review by louis-cqrl — APPROVED** [review]
LGTM, same comment as @dustmop

File: pkg/api/util/util.go (hunk 1 above)
Comment (pgimalac) [reply pkg/api/util/util.go:30]: Sure haha, I didn't notice that

**Pull request merged**